awx/main/harakiri_middleware.py

```diff
@@ -0,0 +1,28 @@
+import traceback
```

Comment:
I don't love putting it in the path of `awx/main/harakiri_middleware.py`. That just seems like dumping because of a hesitation to refactor.

A intriguing alternative location (mostly same file otherwise):

https://github.com/ansible/django-ansible-base/tree/devel/ansible_base/lib/middleware

Reply:
@AlanCoding put it there instead of in `middleware.py` to avoid circular import situation that arose when I put it in there in `awx/main/apps.py`